### ISSUE TITLE
Non-portable path to file ".../Appcenter-prefix.pch" specified path differs in case from file name on disk

### ISSUE THREAD

**Issue by yvbeek** [microsoft/appcenter-sdk-react-native#781]
### **Description**

When I compile my project, I'm getting the following warning:
```
/<built-in>:1:10: Non-portable path to file '"<MyProject>/ios/Pods/Target Support
 Files/AppCenter/appcenter-prefix.pch"'; specified path differs in 
case from file name on disk
```

Might be related to:
https://github.com/microsoft/appcenter-sdk-apple/issues/1788

### **Repro Steps**

- Create a new React-Native project
- Use the package.json and Podfile below
- Compile the Xcode project in the ios folder

### **Details**

- App Center SDK version 2.6.0
- CocoaPods version 1.8.4

```
System:
    OS: macOS Mojave 10.14.6
    CPU: (12) x64 Intel(R) Core(TM) i9-8950HK CPU @ 2.90GHz
    Memory: 2.03 GB / 32.00 GB
    Shell: 3.2.57 - /bin/bash
  Binaries:
    Node: 13.1.0 - /usr/local/bin/node
    Yarn: 1.19.2 - /usr/local/bin/yarn
    npm: 6.13.3 - /usr/local/bin/npm
    Watchman: 4.9.0 - /usr/local/bin/watchman
  SDKs:
    iOS SDK:
      Platforms: iOS 13.2, DriverKit 19.0, macOS 10.15, tvOS 13.2, watchOS 6.1
  IDEs:
    Android Studio: 3.5 AI-191.8026.42.35.5977832
    Xcode: 11.3/11C29 - /usr/bin/xcodebuild
  npmPackages:
    react: 16.9.0 => 16.9.0 
    react-native: 0.61.5 => 0.61.5
```

#### **Package.json**

```json
{
  "name": "myproject",
  "private": true,
  "scripts": {
    "start": "react-native start",
    "postinstall": "npx jetify && pod install --project-directory=ios"
  },
  "dependencies": {
    "@react-native-community/async-storage": "^1.7.0",
    "@react-native-community/netinfo": "^5.0.0",
    "@react-native-firebase/analytics": "^6.2.0",
    "@react-native-firebase/app": "^6.2.0",
    "@react-native-firebase/crashlytics": "^6.2.0",
    "appcenter": "^2.6.0",
    "appcenter-analytics": "^2.6.0",
    "base-64": "^0.1.0",
    "expo-av": "8.0.0",
    "expo-secure-store": "8.0.0",
    "expo-web-browser": "8.0.0",
    "lodash.debounce": "^4.0.8",
    "prop-types": "^15.7.2",
    "query-string": "^6.9.0",
    "react": "16.9.0",
    "react-native": "0.61.5",
    "react-native-code-push": "^6.0.0",
    "react-native-extended-stylesheet": "^0.12.0",
    "react-native-gesture-handler": "^1.5.2",
    "react-native-in-app-utils": "^6.0.2",
    "react-native-linear-gradient": "^2.5.6",
    "react-native-material-dropdown": "^0.11.1",
    "react-native-orientation-locker": "^1.1.7",
    "react-native-reanimated": "^1.4.0",
    "react-native-render-html": "^4.1.2",
    "react-native-screens": "^2.0.0-alpha.17",
    "react-native-search-box": "^0.0.19",
    "react-native-sideswipe": "^1.5.0",
    "react-native-unimodules": "0.7.0",
    "react-native-vector-icons": "^6.6.0",
    "react-navigation": "^4.0.10",
    "react-navigation-stack": "^1.10.3",
    "react-navigation-tabs": "^2.6.2",
    "uuid-js": "^0.7.5"
  },
  "devDependencies": {
    "@babel/core": "^7.7.5",
    "@babel/runtime": "^7.7.6",
    "@react-native-community/eslint-config": "^0.0.5",
    "@types/jest": "^24.0.23",
    "@types/react-native": "^0.60.25",
    "@types/react-native-vector-icons": "^6.4.4",
    "@types/react-test-renderer": "16.9.1",
    "@typescript-eslint/eslint-plugin": "^2.11.0",
    "@typescript-eslint/parser": "^2.11.0",
    "babel-jest": "^24.9.0",
    "babel-plugin-module-resolver": "^3.2.0",
    "eslint": "^6.7.2",
    "eslint-plugin-react": "^7.17.0",
    "jest": "^24.9.0",
    "metro-react-native-babel-preset": "^0.57.0",
    "react-test-renderer": "16.9.0",
    "typescript": "^3.7.3"
  },
  "jest": {
    "preset": "react-native",
    "moduleFileExtensions": [
      "ts",
      "tsx",
      "js",
      "jsx",
      "json",
      "node"
    ]
  }
}
```

#### **Podfile**

```ruby
source 'https://github.com/CocoaPods/Specs.git'
platform :ios, '10.0'
inhibit_all_warnings!

require_relative '../node_modules/@react-native-community/cli-platform-ios/native_modules'
require_relative '../node_modules/react-native-unimodules/cocoapods.rb'

project 'MyProject.xcodeproj'

target 'MyProject' do
  pod 'FBLazyVector', :path => "../node_modules/react-native/Libraries/FBLazyVector"
  pod 'FBReactNativeSpec', :path => "../node_modules/react-native/Libraries/FBReactNativeSpec"
  pod 'RCTRequired', :path => "../node_modules/react-native/Libraries/RCTRequired"
  pod 'RCTTypeSafety', :path => "../node_modules/react-native/Libraries/TypeSafety"
  pod 'React', :path => '../node_modules/react-native/'
  pod 'React-Core', :path => '../node_modules/react-native/'
  pod 'React-CoreModules', :path => '../node_modules/react-native/React/CoreModules'
  pod 'React-Core/DevSupport', :path => '../node_modules/react-native/'
  pod 'React-RCTActionSheet', :path => '../node_modules/react-native/Libraries/ActionSheetIOS'
  pod 'React-RCTAnimation', :path => '../node_modules/react-native/Libraries/NativeAnimation'
  pod 'React-RCTBlob', :path => '../node_modules/react-native/Libraries/Blob'
  pod 'React-RCTImage', :path => '../node_modules/react-native/Libraries/Image'
  pod 'React-RCTLinking', :path => '../node_modules/react-native/Libraries/LinkingIOS'
  pod 'React-RCTNetwork', :path => '../node_modules/react-native/Libraries/Network'
  pod 'React-RCTSettings', :path => '../node_modules/react-native/Libraries/Settings'
  pod 'React-RCTText', :path => '../node_modules/react-native/Libraries/Text'
  pod 'React-RCTVibration', :path => '../node_modules/react-native/Libraries/Vibration'
  pod 'React-Core/RCTWebSocket', :path => '../node_modules/react-native/'

  pod 'React-cxxreact', :path => '../node_modules/react-native/ReactCommon/cxxreact'
  pod 'React-jsi', :path => '../node_modules/react-native/ReactCommon/jsi'
  pod 'React-jsiexecutor', :path => '../node_modules/react-native/ReactCommon/jsiexecutor'
  pod 'React-jsinspector', :path => '../node_modules/react-native/ReactCommon/jsinspector'
  pod 'ReactCommon/jscallinvoker', :path => "../node_modules/react-native/ReactCommon"
  pod 'ReactCommon/turbomodule/core', :path => "../node_modules/react-native/ReactCommon"
  pod 'Yoga', :path => '../node_modules/react-native/ReactCommon/yoga'

  pod 'DoubleConversion', :podspec => '../node_modules/react-native/third-party-podspecs/DoubleConversion.podspec'
  pod 'glog', :podspec => '../node_modules/react-native/third-party-podspecs/glog.podspec'
  pod 'Folly', :podspec => '../node_modules/react-native/third-party-podspecs/Folly.podspec'

  # CodePush for over-the-air updates
  pod 'CodePush', :path => '../node_modules/react-native-code-push'

  # Set Firebase packages versions
  $CrashlyticsSDKVersion = '3.14.0'
  $FabricSDKVersion = '1.10.2'
  $FirebaseSDKVersion = '6.13.0'

  use_native_modules!
  use_unimodules!(exclude: [
    'unimodules-barcode-scanner-interface',
    'unimodules-face-detector-interface',
    'unimodules-sensors-interface',
    'unimodules-task-manager-interface'
  ])
end

post_install do |installer|
  installer.pods_project.main_group.tab_width = '2';
  installer.pods_project.main_group.indent_width = '2';

  # Set the deployment target of all pods
  installer.target_installation_results.pod_target_installation_results
    .each do |pod_name, target_installation_result|
    target_installation_result.native_target.build_configurations.each do |config|
      config.build_settings['IPHONEOS_DEPLOYMENT_TARGET'] = '10.0'
    end
  end
end
```

**Comment by Jamminroot** [comment]
Hey! Sorry for such a delay in response.
We have item tracked on our backlog which would fix origin of this issue, but there aren't any ETAs yet I could provide.
I'll leave issue opened and we will keep you updated on the progress in this thread.

**Comment by yvbeek** [comment]
@Jamminroot Any news on this issue?

I did a simple `grep` on my project and I noticed that the casing of `appcenter` in the Pods project is incorrect. That is likely the reason we get this warning:

```bash
./ios/Pods/Pods.xcodeproj/project.pbxproj:
GCC_PREFIX_HEADER = "Target Support Files/appcenter/appcenter-prefix.pch";
./ios/Pods/Pods.xcodeproj/project.pbxproj:
GCC_PREFIX_HEADER = "Target Support Files/appcenter/appcenter-prefix.pch";
./ios/Pods/Pods.xcodeproj/project.pbxproj:
GCC_PREFIX_HEADER = "Target Support Files/appcenter/appcenter-prefix.pch";
```

The correct casing is: `Target Support Files/AppCenter/appcenter-prefix.pch`
You can probably fix this by setting the name of the Pod correctly:

By changing

```ruby
Pod::Spec.new do |s|
  s.name              = package['name']
```

to 

```ruby
Pod::Spec.new do |s|
  s.name              = 'AppCenter'
```

**Comment by annakocheshkova** [comment]
Hi there. This has been fixed and should be in one of the next releases.

**Comment by guperrot** [comment]
The new version of the SDK has been released with a fix: [3.0.0](https://github.com/microsoft/appcenter-sdk-react-native/releases/tag/3.0.0).
